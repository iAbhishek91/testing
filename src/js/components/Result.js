import React from 'react';
import PropTypes from 'prop-types';

const Result = props => {
  return <p className="result">{props.result}</p>;
};

Result.defaultProps = {
  result: '0'
};

Result.propTypes = {
  result: PropTypes.string.isRequired
};

export default Result;
