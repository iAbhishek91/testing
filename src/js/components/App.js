import React, { Component, Fragment } from 'react';
import ReactDom from 'react-dom';
import math from 'mathjs';
import Result from './Result';
import Keys from './keys';

class App extends Component {
  constructor() {
    super();
    this.state = {
      title: 'calculator',
      result: '0',
      expression: ''
    };
  }

  calculateClickHandler = () => {
    let evalExpression;
    try {
      evalExpression = math.eval(this.state.expression).toString();
    } catch (err) {
      evalExpression = err.toString();
    }
    this.setState({ result: evalExpression });
  };

  numberClickHandler = key => {
    this.setState({ expression: this.state.expression + key });
  };

  clearHander = () => {
    this.setState({ expression: '', result: 'cleared!!' });
  };

  render() {
    return (
      <Fragment>
        <h1>{this.state.title}</h1>
        <p>{this.state.expression}</p>
        <Result result={this.state.result} />
        <li className="list-style-none">
          <Keys
            type="number"
            handler={this.numberClickHandler.bind(this, '1')}
            value="1"
          />
          <Keys
            type="number"
            handler={this.numberClickHandler.bind(this, '2')}
            value="2"
          />
          <Keys
            type="number"
            handler={this.numberClickHandler.bind(this, '3')}
            value="3"
          />
          <Keys type="clear" handler={this.clearHander} value="C" />
        </li>
        <li className="list-style-none">
          <Keys
            type="number"
            handler={this.numberClickHandler.bind(this, '4')}
            value="4"
          />
          <Keys
            type="number"
            handler={this.numberClickHandler.bind(this, '5')}
            value="5"
          />
          <Keys
            type="number"
            handler={this.numberClickHandler.bind(this, '6')}
            value="6"
          />
          <Keys
            type="operation"
            handler={this.numberClickHandler.bind(this, '+')}
            value="+"
          />
        </li>
        <li className="list-style-none">
          <Keys
            type="number"
            handler={this.numberClickHandler.bind(this, '7')}
            value="7"
          />
          <Keys
            type="number"
            handler={this.numberClickHandler.bind(this, '8')}
            value="8"
          />
          <Keys
            type="number"
            handler={this.numberClickHandler.bind(this, '9')}
            value="9"
          />
          <Keys
            type="operation"
            handler={this.numberClickHandler.bind(this, '-')}
            value="-"
          />
        </li>
        <li className="list-style-none">
          <Keys
            type="number"
            handler={this.numberClickHandler.bind(this, '.')}
            value="."
          />
          <Keys
            type="number"
            handler={this.numberClickHandler.bind(this, '0')}
            value="0"
          />
          <Keys
            type="operation"
            handler={this.numberClickHandler.bind(this, '/')}
            value="/"
          />
          <Keys
            type="operation"
            handler={this.numberClickHandler.bind(this, '*')}
            value="*"
          />
        </li>
        <li className="list-style-none">
          <Keys
            type="equalTo"
            id="equalTo"
            handler={this.calculateClickHandler}
            value="="
          />
        </li>
      </Fragment>
    );
  }
}

export default App;
