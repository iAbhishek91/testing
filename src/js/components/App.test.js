import App from './App';
import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Result from './Result';
import renderer from 'react-test-renderer';

configure({ adapter: new Adapter() });

it("test 'Result' component is rendered once", () => {
  const renderApp = shallow(<App />);
  expect(renderApp.find(Result)).toHaveLength(1);
});

it('test snapshot of <App /> componenet', () => {
  let tree = renderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});

it.skip("test 'Result' component is rendered once", () => {
  const renderApp = shallow(<App />);
  expect(renderApp.find(Result)).toHaveLength(1);
});
