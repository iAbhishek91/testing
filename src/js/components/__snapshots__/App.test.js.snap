// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test snapshot of <App /> componenet 1`] = `
Array [
  <h1>
    calculator
  </h1>,
  <p>
    
  </p>,
  <p
    className="result"
  >
    0
  </p>,
  <li
    className="list-style-none"
  >
    <button
      className="number"
      onClick={[Function]}
      type="button"
    >
      1
    </button>
    <button
      className="number"
      onClick={[Function]}
      type="button"
    >
      2
    </button>
    <button
      className="number"
      onClick={[Function]}
      type="button"
    >
      3
    </button>
    <button
      className="clear"
      onClick={[Function]}
      type="button"
    >
      C
    </button>
  </li>,
  <li
    className="list-style-none"
  >
    <button
      className="number"
      onClick={[Function]}
      type="button"
    >
      4
    </button>
    <button
      className="number"
      onClick={[Function]}
      type="button"
    >
      5
    </button>
    <button
      className="number"
      onClick={[Function]}
      type="button"
    >
      6
    </button>
    <button
      className="operation"
      onClick={[Function]}
      type="button"
    >
      +
    </button>
  </li>,
  <li
    className="list-style-none"
  >
    <button
      className="number"
      onClick={[Function]}
      type="button"
    >
      7
    </button>
    <button
      className="number"
      onClick={[Function]}
      type="button"
    >
      8
    </button>
    <button
      className="number"
      onClick={[Function]}
      type="button"
    >
      9
    </button>
    <button
      className="operation"
      onClick={[Function]}
      type="button"
    >
      -
    </button>
  </li>,
  <li
    className="list-style-none"
  >
    <button
      className="number"
      onClick={[Function]}
      type="button"
    >
      .
    </button>
    <button
      className="number"
      onClick={[Function]}
      type="button"
    >
      0
    </button>
    <button
      className="operation"
      onClick={[Function]}
      type="button"
    >
      /
    </button>
    <button
      className="operation"
      onClick={[Function]}
      type="button"
    >
      *
    </button>
  </li>,
  <li
    className="list-style-none"
  >
    <button
      className="equalTo"
      id="equalTo"
      onClick={[Function]}
      type="button"
    >
      =
    </button>
  </li>,
]
`;
